packages/kit/src/runtime/client/client.js

```diff
@@ -610,6 +636,26 @@ export function create_client(app, target) {
 		return null;
 	}
 
+	/**
+	 *
+	 * @param {URL} [old_url]
+	 * @param {URL} [new_url]
+	 */
+	function check_search_params_changed(old_url, new_url) {
+		const changed = new Set();
+		const new_search_params = new URLSearchParams(new_url?.searchParams);
+		for (const [key, value] of old_url?.searchParams?.entries?.() ?? []) {
+			if (new_search_params.get(key) !== value) {
+				changed.add(key);
+			}
+			new_search_params.delete(key);
```

Comment:
Do we need to do the "create copy and delete" dance here? I suspect you do this because there can be multiple parameters with the same key - but `new_search_params.get(key)` will always return the first value, so if there are multiple, this does result in changed parameters anyway. I think we can simplify this a bit here (and have a bit less code as a consequence)

Reply:
No this was because you also want to rerun if a new searchParam is added so basically if you go from

`/?test=true` to `/?test=true&something=else` we want to add something to the list of changed parameters. So when the for finishes everything that is still in the `new_search_params` is effectively "changed"

Reply:
What about using .getAll() and comparing the array values?

Reply:
How does it solve this issue? Even if i `getAll` i will never check if `new_search_params.getAll("something")` because something it's not in `old_search_params`. If i loop over `new_search_params` the same applies to the removal of a searchParam.

Reply:
However now i wonder if we should still use getAll to ensure that we check all the query params because i think delete will delete every key. So we could miss something like

`/?test=1&test=2` to `/?test=1&test=3`

is that what you were saying?

Reply:
Confirmed...i'l fix this

Reply:
Yeah, sorry I should have created a new thread. I just commented on this one without context 😬

Reply:
@dummdidumm fixed this in the new commit